tests/sources/test_redis.py

```diff
@@ -213,3 +218,105 @@ async def test_get_db_records():
         source.client.get_paginated_key = AsyncIterator(["0"])
         async for record in source.get_db_records(db=0):
             assert record == DOCUMENT[0]
+
+
+@pytest.mark.parametrize(
+    "filtering",
+    [
+        Filter(
+            {
+                ADVANCED_SNIPPET: {
+                    "value": [
+                        {"database": 0},
+                    ]
+                }
+            }
+        ),
+    ],
+)
+@pytest.mark.asyncio
+@freeze_time("2023-01-24T04:07:19+00:00")
+async def test_get_docs_with_sync_rules(filtering):
+    async with create_redis_source() as source:
+        source.client.database = ["*"]
+        source.get_db_records = AsyncIterator(items=DOCUMENT)
+        async for (doc, _) in source.get_docs(filtering):
+            assert doc in DOCUMENT
```

Comment:
This is not testing anything. At least you should mock a `RedisClient`, and assert `get_paginated_key` is called with the `database`, `pattern` and `_type` of the rule (you should also add `patten` and/or `_type` in the rule.